platform/server/elasticsearch/__tests__/ScopedDataClient.test.ts

```diff
@@ -0,0 +1,49 @@
+const mockCallAPI = jest.fn();
+
+jest.mock('../call_api', () => ({
+  callAPI: mockCallAPI,
+}));
+
+import { Client } from 'elasticsearch';
+import { ElasticsearchConfig, ElasticsearchClusterType } from '../ElasticsearchConfig';
+import * as clusterSchema from '../schema';
+import { ScopedDataClient } from '../ScopedDataClient';
+
+let client;
+let esClient: Client;
+let esConfig: ElasticsearchConfig;
+let clusterType: ElasticsearchClusterType;
+
+beforeEach(() => {
+  esClient = new Client({});
+  clusterType = 'data';
+  esConfig = new ElasticsearchConfig(clusterType, clusterSchema);
```

Comment:
My naming in the elasticsearch config is a bit confusing, so this is injecting the wrong thing as the last param. It's supposed to be the "raw elasticsearch config" (aka what we've received from the yaml file and which has been validated by the schema).

Here's a suggestion to fix this:

```diff
diff --git a/platform/server/elasticsearch/__tests__/ScopedDataClient.test.ts b/platform/server/elasticsearch/__tests__/ScopedDataClient.test.ts
index 90c3720fb..2e1c0e0d8 100644
--- a/platform/server/elasticsearch/__tests__/ScopedDataClient.test.ts
+++ b/platform/server/elasticsearch/__tests__/ScopedDataClient.test.ts
@@ -5,20 +5,19 @@ jest.mock('../call_api', () => ({
 }));
 
 import { Client } from 'elasticsearch';
-import { ElasticsearchConfig, ElasticsearchClusterType } from '../ElasticsearchConfig';
-import * as clusterSchema from '../schema';
+import { ElasticsearchConfig } from '../ElasticsearchConfig';
 import { ScopedDataClient } from '../ScopedDataClient';
 
-let client;
+let client: ScopedDataClient;
 let esClient: Client;
 let esConfig: ElasticsearchConfig;
-let clusterType: ElasticsearchClusterType;
 
 beforeEach(() => {
   esClient = new Client({});
-  clusterType = 'data';
-  esConfig = new ElasticsearchConfig(clusterType, clusterSchema);
-  esConfig.requestHeadersWhitelist = ['authorization'];
+  const rawEsConfig: any = {
+    requestHeadersWhitelist: ['authorization']
+  };
+  esConfig = new ElasticsearchConfig('data', rawEsConfig);
   client = new ScopedDataClient({
     client: esClient,
     headers: { authorization: 'auth', foo: 'bar' },
```